src/Microsoft.Extensions.Options/IOptionsValidator.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Used to validate TOptions instances.
+    /// </summary>
+    /// <typeparam name="TOptions">The type of options being requested.</typeparam>
+    public interface IOptionsValidator<TOptions> where TOptions : class, new()
```

Comment:
For now that's fine.

According my PR (https://github.com/aspnet/Options/pull/171) there are some changes that might be required to improve validation feature. But it definitely might be applied over this PR, so that's ok.